projects/noloan/common/src/main/java/hasadna/noloan/common/FirebaseAuthentication.java

```diff
@@ -20,20 +26,65 @@ public static FirebaseAuthentication getInstance() {
     return instance;
   }
 
-  public void signinAnonymusly() {
+  public FirebaseAuthentication()
+  {
     auth = FirebaseAuth.getInstance();
+    user = auth.getCurrentUser();
+
+  }
+
+  public void signinAnonymusly() {
     auth.signInAnonymously()
-        .addOnCompleteListener(
-            task -> {
-              if (task.isSuccessful()) {
-                user = auth.getCurrentUser();
-              } else {
-                Log.w(TAG, "signInAnonymously:failure", task.getException());
-              }
-            });
+        .addOnCompleteListener(task -> {
+          if (task.isSuccessful()) {
+            user = auth.getCurrentUser();
+          } else {
+            Log.w(TAG, "signInAnonymously:failure", task.getException());
+          }
+        });
+  }
+
+
+  public TaskCompletionSource<Boolean> signinAdmin(String email, String password) {
+    Log.d(TAG, "starting to log in!");
+    TaskCompletionSource t = new TaskCompletionSource<>();
+
+    //if not already signed in.
```

Comment:
Please add space: `//if->// if`

Also, please start comments, logs and toasts with Capital letters.